### PR TITLE
Update to solid-js 1.8.17, and solidjs/router 0.13.3

### PR DESCRIPTION
**Pull request by goswinr**
I updated to the latest Solid-Js version and fixed a few resulting syntax changes

### CODE REVIEW

File: .gitignore
Comment (goswinr) [inline .gitignore:1]: Simplified gitignore to minimum, is that OK ?

File: index.html
Comment (goswinr) [inline index.html:1]: Changed folder because "dist" seemed to be ignored by Vite watcher

File: src/App/App.fs
Comment (goswinr) [inline src/App/App.fs:1]: Moved out of file App.fs and rewritten to new syntax in Solid JS router

File: src/App/Components.fs [hunk 1]
```diff
@@ -8,17 +8,21 @@ module Color =
...
         Color.context.Provider((color, setColor), children)
 
-    [<JSX.Component>]
-    static member Div (classes: string seq) children =
-        Html.div [
-            Attr.classList classes
-            Html.children children
-        ]
+
+    // does not work anymore(2024), generates invalid JSX: <Components_Div classes={["column", "is-4"]}>
```
Comment (goswinr) [inline src/App/Components.fs:18]: I am not sure why, but this component compiles to invalid JSX, `<Components_Div classes={["column", "is-4"]}>`

File: src/App/Counter.fs
Comment (goswinr) [inline src/App/Counter.fs:1]: just formatted by fantomas

File: src/App/Routing.fs
Comment (goswinr) [inline src/App/Routing.fs:1]: Moved out of file App.fs and rewritten to new syntax in Solid JS router

File: src/App/Solid.Router.fs
Comment (goswinr) [inline src/App/Solid.Router.fs:1]: new solidjs router

File: src/App/TodoElmish.fs
Comment (goswinr) [inline src/App/TodoElmish.fs:1]: Removed Div component, 
see https://github.com/fable-compiler/Fable.Solid/pull/17/files#r1624900819
 formatted by fantomas

**Review by goswinr — COMMENTED** [review]
fixes to restore original behavior from 2022

**Comment by MangelMaxime** [comment]
@goswinr I never used SolidJS, so I am not expert of it.

Would you like me to make you a contributor / provides NuGet permission to make release of the packages?

Of course, I will be available to answer your questions and provide feedback but like that I have one less repository to focus on for release etc. and it will probably make it easier to get new releases. 😇

File: .gitignore
Comment (MangelMaxime) [reply .gitignore:1]: I have no problem with that, I also keep my `.gitignore` file minimal in general.

**Comment by MangelMaxime** [comment]
Regarding Fantomas here is the configuration I am using now days.

The `[demo/**/*.{fs,fsx}]` section is specific to project that are using the Feliz syntax style. If you think it can improve the readability feel free to adopt it.

```yaml
[*.{fs,fsx}]
max_line_length = 100
fsharp_alternative_long_member_definitions = true
fsharp_multi_line_lambda_closing_newline = true
fsharp_multiline_bracket_style = aligned
fsharp_keep_max_number_of_blank_lines = 1
fsharp_align_function_signature_to_indentation = true
fsharp_max_if_then_else_short_width = 0

[demo/**/*.{fs,fsx}]
fsharp_experimental_elmish = true
fsharp_record_multiline_formatter = number_of_items
fsharp_array_or_list_multiline_formatter = number_of_items
```

**Comment by goswinr** [comment]
Thanks for your trust @MangelMaxime , I also just started to try out SolidJs. So the first step was to get the samples running with the latest version.  I like it so far. yes, I would be happy to be a contributor and publish packages.

**Comment by goswinr** [comment]
Replaced by https://github.com/fable-compiler/Fable.Solid/pull/18

**Pull request closed without merging**